src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpServer.cs

```diff
@@ -206,14 +206,18 @@ private static void OnDisconnect(IntPtr pvManagedHttpContext)
             IISHttpContext context = null;
             try
             {
-                context = (IISHttpContext)GCHandle.FromIntPtr(pvManagedHttpContext).Target;
+                var gcHandle = GCHandle.FromIntPtr(pvManagedHttpContext);
+                if (!gcHandle.IsAllocated)
```

Comment:
Why?

Reply:
If the gcHandle is null, we will do a needless throw/catch of a NullReferenceException.

Reply:
Why do we ever call OnDisconnect with null handle? I though new logic in native should've fixed it.